### ISSUE TITLE
Defines in main language file should not be redefined in includes/languages/english/<page>.php

### ISSUE THREAD

**Issue by scottcwilson** [zencart/zencart#2379]
./languages/english.php:  define('TABLE_HEADING_QUANTITY', 'Qty.');
./languages/english/advanced_search.php:  define('TABLE_HEADING_QUANTITY', 'Quantity');
./languages/english/advanced_search_result.php:define('TABLE_HEADING_QUANTITY', 'Quantity');

The second and third defines will fail because the first one is done so early (when application_top is included.) 

This define needs to be removed from files in the english folder and only defined once in the main language file.

**Comment by mc12345678** [comment]
This issue makes me wonder why the language system is currently designed to not offer true override capability? Currently, if it is desired for a page to use a different definition than what is provided in the main language file, then the main language file must be modified along with the language file for all other pages that use/might use the define that was to be overridden or the template file for the associated page and language file for that file needs to be modified to use a different define that is then added to the page's language file(s).

Basically, as is anything in `includes/languages/LANGUAGE.php` can only be "overridden" by `includes/languages/YOUR_TEMPLATE/LANGUAGE.php` 

but not by 
`includes/languages/LANGUAGE/YOUR_PAGE.php` nor `includes/languages/LANGUAGE/YOUR_TEMPLATE/YOUR_PAGE.php`

Is there a technical reason for this? Is it expected to change in a future significant release (I can't say that I saw anything in the develop branch that seemed to modify that approach)?

**Comment by scottcwilson** [comment]
Fair question - I am guessing this is a legacy constraint.  Right now the main language file is loaded first, so it takes priority.  Changing this could break things for upgraders - not sure it's wise.  I have been arguing for just consolidating these defines.  Do you REALLY need different values for a string like Quantity?

**Comment by drbyte** [comment]
> Do you REALLY need different values for a string like Quantity?

Well, when a column heading is being shortened to just `Qty` to not waste horizontal space, yes.

**Comment by torvista** [comment]
Why not just have:
> ./languages/english.php: define('TABLE_HEADING_QTY', 'Qty.');
> ./languages/english.php: define('TABLE_HEADING_QUANTITY', 'Quantity');

and remove both defines from the page-specific lang file.

If a busy table uses the abbreviation out of the box, there will be no need to change it.
If a sparse table using the complete heading gets modifed such that it needs the abbreviation...well so what, it gets modified in the script anyway.

**Comment by lankeeyankee** [comment]
There are a decent amount of these in the admin and store side. For instance, there are several duplicates throwing notices in the admin main english file and attributes controller language file.  

Is this the type of thing where one just does PRs as they come across them or is there a planned effort to "do a sweep" through the files and get them all at once?

**Comment by scottcwilson** [comment]
Sit tight for a little bit ... this is a controversial area where a direction needs to be set.  Then you can go in and pick off a bunch.

**Comment by drbyte** [comment]
The admin-side language stuff is getting some tweaks.
The catalog-side loading of page files by wildcard is intentional. Clashes can be avoided where they make sense.
